### PR TITLE
Viewer/Editor syntax style override

### PR DESCRIPTION
**Pull request by pskowronek**
Added a View menu item to override syntax style (the list is auto-generated from RSyntaxTextArea supported syntax styles).

### CODE REVIEW

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java [hunk 1]
```diff
@@ -311,8 +312,42 @@ void selectAll() {
...
+        for (Field field : fields) {
+            String name = field.getName();
+            if (name.startsWith("SYNTAX_STYLE_")) {
+                String[] parts = name.substring("SYNTAX_STYLE_".length()).split("_");
+                StringBuilder prettyName = new StringBuilder(parts[0]);
+                if (parts.length > 1) {
+                    for (int i = 1; i < parts.length; i++) {
+                        prettyName.append(" ");
+                        prettyName.append(parts[i].toLowerCase());
+                    }
```
Comment (ahadas) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java:328]: I didn't try that but seems we can replace 6 line with a single one like:
```java
Stream.of(parts).map(String::toLowerCase).collect(Collectors.joining(" "))
```
no?

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java [hunk 2]
```diff
@@ -311,8 +312,42 @@ void selectAll() {
...
+                }
+                syntaxList.add(prettyName.toString());
+            }
+        }
+        return syntaxList;
+    }
+
+    void setSyntaxStyle(String syntaxStyleHuman) {
+        // TODO use enum when/if https://github.com/bobbylight/RSyntaxTextArea/issues/479 implemented
+        String normalizedNAme = "SYNTAX_STYLE_" + syntaxStyleHuman.replace(" ", "_").toUpperCase();
```
Comment (ahadas) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java:338]:
```suggestion
        String normalizedName = "SYNTAX_STYLE_" + syntaxStyleHuman.replace(" ", "_").toUpperCase();
```

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java [hunk 3]
```diff
@@ -311,8 +312,42 @@ void selectAll() {
...
+        Field[] fields = SyntaxConstants.class.getFields();
+        for (Field field : fields) {
+            String name = field.getName();
+            if (normalizedNAme.equals(name)) {
```
Comment (ahadas) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java:342]:
```suggestion
            if (normalizedName.equals(name)) {
```

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java [hunk 4]
```diff
@@ -239,13 +239,28 @@ protected void initMenuBarItems() { // TODO code dup with TextEditor, fix it
...
         // View menu
         viewMenu = new JMenu(Translator.get("text_viewer.view"));
 
         JMenuItem item;
 
+        JMenu tabSyntaxMenu = new JMenu(Translator.get("text_viewer.syntax_style"));
+
+        ButtonGroup group = new ButtonGroup();
+        for (String syntaxStyle : textEditorImpl.getSyntaxStyles()) {
+            // TODO reflect the current style (that was automagically set on open)
```
Comment (ahadas) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java:251]: +1

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java [hunk 5]
```diff
@@ -239,13 +239,28 @@ protected void initMenuBarItems() { // TODO code dup with TextEditor, fix it
...
+            JRadioButtonMenuItem radio = new JRadioButtonMenuItem(syntaxStyle, false);
+            radio.addActionListener(
+                    e -> {
+                        textEditorImpl.setSyntaxStyle(syntaxStyle);
+                    }
+            );
```
Comment (ahadas) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java:257]: how about:
```java
radio.addActionListener(e -> textEditorImpl.setSyntaxStyle(syntaxStyle));
```

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java [hunk 6]
```diff
@@ -133,6 +133,25 @@ protected void initMenuBarItems() { // TODO code dup with TextViewer, fix it
...
+                for (String syntaxStyle : textEditorImpl.getSyntaxStyles()) {
+                    // TODO reflect the current style (that was automagically set on open)
+                    JRadioButtonMenuItem radio = new JRadioButtonMenuItem(syntaxStyle, false);
+                    radio.addActionListener(
+                            e -> {
+                                boolean saveNeededKeeper = isSaveNeeded();
+                                textEditorImpl.setSyntaxStyle(syntaxStyle);
+                                setSaveNeeded(saveNeededKeeper);
+                            }
+                    );
```
Comment (ahadas) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java:148]: same comment as below on TextViewer

**Review by ahadas — COMMENTED** [review]
looks good, minor comments inside

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java (hunk 6 above)
Comment (pskowronek) [reply mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java:148]: not sure if it is relevant, or I don't get it :)

**Comment by pskowronek** [comment]
PR updated

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java (hunk 6 above)
Comment (ahadas) [reply mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java:148]: I'm not sure now what I meant by that either :)

**Review by ahadas — APPROVED** [review]

**Pull request merged**